### PR TITLE
[bitnami/postgresql, zookeeper]

### PR DESCRIPTION
**Pull request by bbaudelet**
**Description of the change**

- postgresql

Add template render for extraVolumes on sts and for the additionalLabels on the serviceMonitor.

- zookeeper

Add support of ZOO_LOG4J_PROP env properties to 

Add sidecars parameters to add another container to  pod
Add template render for extraVolumes.

**Benefits**

This modification makes possible to use some variables such as Release.Name on ExtraVolumes part.
as exemple it allows the user to mount configmaps which may have a name linked to  release name.

- zookeeper

User can add any other container needed on the zookeeper pod as log exporter.

**Possible drawbacks**
No drawbacks, no impact on current usage. 

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/postgresql/README.md [hunk 1]
```diff
@@ -210,7 +210,7 @@ $ kubectl delete pvc -l release=my-release
...
 | `primary.extraInitContainers`                 | Extra init containers to add to the pods (postgresql primary)                                                                                             | `[]`                        |
 | `primary.extraVolumeMounts`                   | Extra volume mounts to add to the pods (postgresql primary)                                                                                               | `[]`                        |
-| `primary.extraVolumes`                        | Extra volumes to add to the pods (postgresql primary)                                                                                                     | `[]`                        |
+| `primary.extraVolumes`                        | Extra volumes to add to the pods (postgresql primary), the value is evaluated as a template.                                                                                                    | `[]`                        |
```
Comment (alvneiayu) [inline bitnami/postgresql/README.md:213]: I think that you don't need to include this in the Readme. I took a look another charts like Redis and we are not including this information. So, please remove it. Thanks.

File: bitnami/postgresql/README.md [hunk 2]
```diff
@@ -230,7 +230,7 @@ $ kubectl delete pvc -l release=my-release
...
 | `readReplicas.extraInitContainers`            | Extra init containers to add to the pods (postgresql readReplicas)                                                                                        | `[]`                        |
 | `readReplicas.extraVolumeMounts`              | Extra volume mounts to add to the pods (postgresql readReplicas)                                                                                          | `[]`                        |
-| `readReplicas.extraVolumes`                   | Extra volumes to add to the pods (postgresql readReplicas)                                                                                                | `[]`                        |
+| `readReplicas.extraVolumes`                   | Extra volumes to add to the pods (postgresql readReplicas), the value is evaluated as a template.                                                                                               | `[]`                        |
```
Comment (alvneiayu) [inline bitnami/postgresql/README.md:233]: I think that you don't need to include this in the Readme. I took a look another charts like Redis and we are not including this information. So, please remove it. Thanks.

File: bitnami/postgresql/README.md [hunk 3]
```diff
@@ -276,7 +276,7 @@ $ kubectl delete pvc -l release=my-release
...
 | `metrics.service.loadBalancerIP`              | loadBalancerIP if redis metrics service type is `LoadBalancer`                                                                                            | `""`                        |
 | `metrics.serviceMonitor.enabled`              | Set this to `true` to create ServiceMonitor for Prometheus operator                                                                                       | `false`                     |
-| `metrics.serviceMonitor.additionalLabels`     | Additional labels that can be used so ServiceMonitor will be discovered by Prometheus                                                                     | `{}`                        |
+| `metrics.serviceMonitor.additionalLabels`     | Additional labels that can be used so ServiceMonitor will be discovered by Prometheus  , the value is evaluated as a template.                                                                   | `{}`                        |
```
Comment (alvneiayu) [inline bitnami/postgresql/README.md:279]: I think that you don't need to include this in the Readme. I took a look another charts like Redis and we are not including this information. So, please remove it. Thanks.

File: bitnami/zookeeper/README.md [hunk 4]
```diff
@@ -111,7 +111,7 @@ The command removes all the Kubernetes components associated with the chart and
 
 | Name                                 | Description                                                                                                                                                                                       | Value           |
 | ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------- |
-| `extraVolumes`                       | Extra volumes                                                                                                                                                                                     | `[]`            |
+| `extraVolumes`                       | Extra volumes  (value evaluated as a template)                                                                                                                                                                                     | `[]`            |
```
Comment (alvneiayu) [inline bitnami/zookeeper/README.md:114]: I think that you don't need to include this in the Readme. I took a look another charts like Redis and we are not including this information. So, please remove it. Thanks.

File: bitnami/zookeeper/templates/statefulset.yaml [hunk 5]
```diff
@@ -291,7 +291,7 @@ spec:
             - name: ZOO_SNAPCOUNT
               value: {{ .Values.snapCount | quote }}
             - name: ZOO_MAX_CLIENT_CNXNS
-              value: {{ .Values.maxClientCnxns | quote }}
+              value: {{ .Values.maxClientCnxns | quote }}
```
Comment (alvneiayu) [inline bitnami/zookeeper/templates/statefulset.yaml:294]: Remove the empty spaces at the end of the line, please

File: bitnami/zookeeper/values.yaml [hunk 6]
```diff
@@ -442,6 +442,18 @@ serviceAccount:
 ## Network policies
 ## Ref: https://kubernetes.io/docs/concepts/services-networking/network-policies/
 ##
+
+sidecar: []
+  ## @param sidecars Extra containers to the pod
```
Comment (alvneiayu) [inline bitnami/zookeeper/values.yaml:447]: Please move the comments before the sidecar key definition (line 447-455 to 445). This must fix the linter Readme fail that you are having.

**Review by alvneiayu — CHANGES_REQUESTED** [review]
Hi @bbaudelet

First of all, thanks a lot for your PR and your time.

I included several comments in the PR. Moreover, you are having linter errors in your PR. I included a comment inside the code to fix one of them but the other one must be fixed bumping the Zookeeper version as you did in Postgresql. 

Please, could you edit the PR title, right now is showing only "[bitnami/postgresql]". Please, edit it to include more information and the zookeeper chart.

This is a personal suggestion for the future, send two PRs separated in the future (one for Postgresql and one for Zookeeper) but this is a personal opinion. I think that it will be more easy for you for describing the changes with a title and the description with two different PRs.

So, let me know if I can help you with something.

Thanks a lot

Álvaro

**Comment by bbaudelet** [comment]
Thanks a lot for your reviewing, i'm sorry to have send only one PR for postgres and zookeeper it's my bad.
For the next PR i send one by chart.

File: bitnami/zookeeper/values.yaml [hunk 7]
```diff
@@ -442,6 +442,19 @@ serviceAccount:
...
+## @param sidecars Extra containers to the pod
+## For example:
+## sidecars:
+##   - name: your-image-name
+##     image: your-image
+##     imagePullPolicy: Always
+##     ports:
+##       - name: portname
+##         containerPort: 1234
+sidecar: []
```
Comment (alvneiayu) [inline bitnami/zookeeper/values.yaml:455]: The key is sidecars, please change it from sidecar to sidecars.

File: bitnami/zookeeper/README.md [hunk 8]
```diff
@@ -111,7 +111,7 @@ The command removes all the Kubernetes components associated with the chart and
 
 | Name                                 | Description                                                                                                                                                                                       | Value           |
 | ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------- |
-| `extraVolumes`                       | Extra volumes                                                                                                                                                                                     | `[]`            |
+| `extraVolumes`                       | Extra volumes.                                                                                                                                                                                     | `[]`            |
```
Comment (alvneiayu) [inline bitnami/zookeeper/README.md:114]: this point is not necessary. Please delete it. Please, leave the line as the original one (no changes, please).

File: bitnami/postgresql/README.md [hunk 9]
```diff
@@ -276,7 +276,7 @@ $ kubectl delete pvc -l release=my-release
 | `metrics.service.annotations`                 | Additional annotations for metrics exporter pod                                                                                                           | `{}`                        |
 | `metrics.service.loadBalancerIP`              | loadBalancerIP if redis metrics service type is `LoadBalancer`                                                                                            | `""`                        |
 | `metrics.serviceMonitor.enabled`              | Set this to `true` to create ServiceMonitor for Prometheus operator                                                                                       | `false`                     |
-| `metrics.serviceMonitor.additionalLabels`     | Additional labels that can be used so ServiceMonitor will be discovered by Prometheus                                                                     | `{}`                        |
+| `metrics.serviceMonitor.additionalLabels`     | Additional labels that can be used so ServiceMonitor will be discovered by Prometheus.                                                                   | `{}`                        |
```
Comment (alvneiayu) [inline bitnami/postgresql/README.md:279]: this point is not necessary. Please delete it. Please, leave the line as the original one (no changes, please).

File: bitnami/postgresql/README.md [hunk 10]
```diff
@@ -230,7 +230,7 @@ $ kubectl delete pvc -l release=my-release
 | `readReplicas.priorityClassName`              | Priority Class to use for each pod (postgresql readReplicas)                                                                                              | `""`                        |
 | `readReplicas.extraInitContainers`            | Extra init containers to add to the pods (postgresql readReplicas)                                                                                        | `[]`                        |
 | `readReplicas.extraVolumeMounts`              | Extra volume mounts to add to the pods (postgresql readReplicas)                                                                                          | `[]`                        |
-| `readReplicas.extraVolumes`                   | Extra volumes to add to the pods (postgresql readReplicas)                                                                                                | `[]`                        |
+| `readReplicas.extraVolumes`                   | Extra volumes to add to the pods (postgresql readReplicas).                                                                                               | `[]`                        |
```
Comment (alvneiayu) [inline bitnami/postgresql/README.md:233]: this point is not necessary. Please delete it. Please, leave the line as the original one (no changes, please).

File: bitnami/postgresql/README.md [hunk 11]
```diff
@@ -210,7 +210,7 @@ $ kubectl delete pvc -l release=my-release
 | `primary.priorityClassName`                   | Priority Class to use for each pod (postgresql primary)                                                                                                   | `""`                        |
 | `primary.extraInitContainers`                 | Extra init containers to add to the pods (postgresql primary)                                                                                             | `[]`                        |
 | `primary.extraVolumeMounts`                   | Extra volume mounts to add to the pods (postgresql primary)                                                                                               | `[]`                        |
-| `primary.extraVolumes`                        | Extra volumes to add to the pods (postgresql primary)                                                                                                     | `[]`                        |
+| `primary.extraVolumes`                        | Extra volumes to add to the pods (postgresql primary).                                                                                                    | `[]`                        |
```
Comment (alvneiayu) [inline bitnami/postgresql/README.md:213]: this point is not necessary. Please delete it. Please, leave the line as the original one (no changes, please).

File: bitnami/zookeeper/values.yaml [hunk 12]
```diff
@@ -442,6 +442,19 @@ serviceAccount:
...
+
+
```
Comment (alvneiayu) [inline bitnami/zookeeper/values.yaml:457]: Remove one new line. This is a cosmetic change.

**Review by alvneiayu — CHANGES_REQUESTED** [review]
Hi @bbaudelet 

Thanks again for your time including the changes. I requested more changes. Moreover, you are having linter problems. To fix it, you must bump the PostgreSQL Chart.yaml version.

Thanks a lot again

Álvaro

**Review by alvneiayu — APPROVED** [review]
LGTM

**Pull request merged**